allennlp/nn/decoding/decoder_state.py

```diff
@@ -0,0 +1,30 @@
+from typing import List
+
+import torch
+
+
+class DecoderState:
+    def __init__(self,
```

Comment:
Can you add a docstring for this class? I know this is still WIP, but I'd like to understand why you're storing the things you're storing in this class.